agent/util-scripts/pbench-tool-meister-start

```diff
@@ -244,13 +204,635 @@ def waitpid(pid):
+
+class RedisServer(BaseServer):
+    """RedisServer - an encapsulation of the handling of the Redis server
+    specification and methods to optionally create and manage an instance.
+    """
+
+    envi_var = "PBENCH_REDIS_SERVER_CFG"
+    def_port = def_redis_port
+    bad_port_ret_code = ReturnCode.BADREDISPORT
+
+    # Redis server configuration template for pbench's use
+    conf_tmpl = """bind {bind_host_names}
+daemonize yes
+dir {tm_dir}
+dbfilename pbench-redis.rdb
+logfile {tm_dir}/redis.log
+loglevel notice
+pidfile {tm_dir}/redis_{redis_port:d}.pid
+port {redis_port:d}
+"""
+
+    def __init__(self, spec):
+        super().__init__()
+        if spec:
+            parts = spec.split(":", 1)
+            if len(parts) != 2:
+                raise self.Err(
+                    f"Bad Redis server specified, {spec!r}", ReturnCode.BADREDISARG
+                )
+            try:
+                self.port = int(parts[1])
+            except Exception as exc:
+                raise self.Err(
+                    f"Bad Redis port specified, {spec!r}", self.bad_port_ret_code
+                ) from exc
+            else:
+                self.host = parts[0]
+            # This instance is created by somebody else.
+            self._managed = False
+        else:
+            self.port = None
+            self.host = None
+            # This instance will be managed by us.
+            self._managed = True
+        self.local_host = None
+        self.pid_file = None
+
+    def start(self, tm_dir, full_hostname, logger):
+        """start_redis - configure and start a Redis server.
+
+        Returns a tuple of the Redis server host name, the host name to be used to
+        connect to it locally, and its port.
+        """
+        assert (
+            self._managed is True
+            and self.host is not None
+            and self.port is not None
+            and self.bind_host is not None
+            and self.bind_port is not None
+        ), f"Logic bomb!  Unexpected state: {self!r}"
+
+        try:
+            bind_host_ip = socket.gethostbyname(self.bind_host)
+        except socket.error as exc:
+            raise self.Err(
+                f"{self.bind_host} does not map to an IP address", ReturnCode.NOIP
+            ) from exc
+        else:
+            assert (
+                bind_host_ip is not None
+            ), f"Logic Bomb!  socket.gethostbyname('{self.bind_host}') returned None"
+        try:
+            host_ip = socket.gethostbyname(self.host)
+        except socket.error as exc:
+            raise self.Err(
+                f"{self.host} does not map to an IP address", ReturnCode.NOIP
+            ) from exc
+        else:
+            assert (
+                host_ip is not None
+            ), f"Logic Bomb!  socket.gethostbyname('{self.host}') returned None"
+            # By default, to talk to the Redis server locally, use the
+            # specified host name.
+            self.local_host = self.host
```

Comment:
I don't think this assignment is ever used -- I think it gets overwritten first.